ui/selectors/permissions.js

```diff
@@ -256,13 +281,37 @@ function getAccountsPermissionFromSubject(subject = {}) {
   return subject.permissions?.eth_accounts || {};
 }
 
+function getChainsFromSubject(subject) {
+  return getChainsFromPermission(getChainsPermissionFromSubject(subject));
+}
+
+function getChainsPermissionFromSubject(subject = {}) {
+  return subject.permissions?.['endowment:permitted-chains'] || {};
```

Comment:
```suggestion
  return subject.permissions?.[PermissionNames.permittedChains] || {};
```

This was marked resolved but I still think we should do it

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/26635/commits/3868d34057e993b1cbcdf570bd5765b62e4bf5e7